driver/parallel.go

```diff
@@ -114,13 +134,76 @@ func (pg *parallelGroup) nextSource() (ScannerCloser, error) {
 	}
 }
 
+// nextSourceForConn is similar to nextSource, but instead of returning the scannerCloser
+// for an open file (i.e. the stream for the open file),
+// nextSourceForConn sends an zapi api request to a remote zqd worker process, and returns
+// the scannerCloser (i.e.output stream) for the remote zqd worker.
+func (pg *parallelGroup) nextSourceForConn(conn *api.Connection) (ScannerCloser, error) {
+	for {
+		select {
+		case src, ok := <-pg.sourceChan:
+			if !ok {
+				return nil, pg.sourceErr
+			}
+
+			req, err := pg.sourceToRequest(src)
+			if err != nil {
+				return nil, err
+			}
+			if req == nil {
+				continue
```

Comment:
sourcetoRequest only returns a nil request when it returns a non-nil err, so this continue is unreachable. Drop it and the for enclosing for loop.